ee/localserver/request-query.go

```diff
@@ -62,6 +64,79 @@ func (ls *localServer) requestQueryHanlderFunc(w http.ResponseWriter, r *http.Re
 
 	w.Write(jsonBytes)
 }
+func (ls *localServer) requestRunScheduledQueryHandler() http.Handler {
+	return http.HandlerFunc(ls.requestRunScheduledQueryHanlderFunc)
+}
+
+func (ls *localServer) requestRunScheduledQueryHanlderFunc(w http.ResponseWriter, r *http.Request) {
+	if r.Body == nil {
+		sendClientError(w, "request body is nil")
+		return
+	}
+
+	var body map[string]any
+	if err := json.NewDecoder(r.Body).Decode(&body); err != nil {
+		sendClientError(w, fmt.Sprintf("error unmarshaling request body: %s", err))
+		return
+	}
+
+	nameRaw, ok := body["name"]
+	if !ok {
+		sendClientError(w, "no name key found in request body json")
+		return
+	}
+
+	name, ok := nameRaw.(string)
+	if !ok {
+		sendClientError(w, fmt.Sprintf("name value not a string: %s", name))
+		return
+	}
+
+	if name == "" {
+		sendClientError(w, "empty name")
+		return
+	}
+
+	scheduledQueryQuery := fmt.Sprintf("select query from osquery_schedule where name = '%s'", name)
+
+	var results []map[string]string
+	var err error
+	backoff.WaitFor(func() error {
+		results, err = ls.querier.Query(scheduledQueryQuery)
+		return err
+	}, 1*time.Second, 250*time.Millisecond)
+
+	if err != nil {
+		sendClientError(w, fmt.Sprintf("error executing query for scheduled query: %s, error: %s", scheduledQueryQuery, err))
+		return
+	}
+
+	if len(results) == 0 {
+		sendClientError(w, fmt.Sprintf("no query found with name '%s' using query %s", name, scheduledQueryQuery))
+		return
+	}
+
+	query, ok := results[0]["query"]
+	if !ok {
+		sendClientError(w, fmt.Sprintf("no query found with name '%s' using query \"%s\"", name, scheduledQueryQuery))
+		return
+	}
+
+	newBody := map[string]string{
+		"query": query,
+	}
+
+	jsonBytes, err := json.Marshal(newBody)
```

Comment:
I think we can drop the check for "query" since thats the only column requested, the length check earlier should handle it, so we can simplify to
```suggestion
	jsonBytes, err := json.Marshal(results[0)
```

Reply:
If you support `LIKE` and running multiple queries, we'll need the query in the response. (And I think we really should support `LIKE`)